src/main/haskell/kore/src/Kore/Step/ExpandedPattern.hs

```diff
@@ -216,9 +217,20 @@ isBottom
   = True
 isBottom _ = False
 
+{-| If an ExpandedPattern is bottom, it is converted to the canonical
+representation of bottom.
+-}
+loBottomize
```

Comment:
I'm not sure why we need to convert things to the canonical bottom, considering we can always call `isBottom`. 

In general, when you have an equivalence class of things, there are two options: either try to convert things to a canonical representation when we produce them, or just test for membership in the class when we consume them. Both work fine but it seems like we've already settled on the latter, so I think we should be consistent and stick to it.

Reply:
That's true, I think... The reason for converting to bottom is to prevent the code that does not use isBottom yet from computing useless stuff. As an example, `stepWithAxiom` only tests the predicate returned by `stepWithAxiom`, and will recursively call itself if only the term is bottom.

I agree that `axiomFunctionEvaluator` shoud use `isBottom`. I agree that it's useless to convert the remainder to bottom. However, the stepWithAxiom function was already converting the result to bottom, and changing that should be in a separate PR, so I will either revert to the old stepWithAxiom code and leave a TODO.